utility-scripts/script/GenerateBLSKeysAndRegisterValidatorsCalldata.s.sol

```diff
@@ -0,0 +1,230 @@
+
+        // 2 token approvals + validator registrations
+        Tx[] memory transactions = new Tx[](numberOfValidators + 2);
+        transactions[0] = Tx({ to: validatorTicketAddress, data: approveVTCalldata });
+        transactions[1] = Tx({ to: address(pufETH), data: approvePufETHCalldata });
+
+        for (uint256 i = 0; i < numberOfValidators; ++i) {
+            // Select the module to register to
+            bytes32 moduleName = moduleWeights[(moduleSelectionIndex + i) % moduleWeights.length];
+
+            // If the user specified a module to register to, use that instead
+            if (moduleToRegisterTo != bytes32(0)) {
+                require(pufferProtocol.getModuleAddress(moduleToRegisterTo) != address(0), "Invalid Puffer Module");
+                moduleName = moduleToRegisterTo;
+            }
+
+            _generateValidatorKey(i, moduleName);
+
+            // Read the registration JSON file
+            registrationJson = vm.readFile(string.concat("./registration-data/", vm.toString(i), ".json"));
+
+            bytes[] memory blsEncryptedPrivKeyShares = new bytes[](guardiansLength);
+            blsEncryptedPrivKeyShares[0] = stdJson.readBytes(registrationJson, ".bls_enc_priv_key_shares[0]");
+
+            ValidatorKeyData memory validatorData = ValidatorKeyData({
+                blsPubKey: stdJson.readBytes(registrationJson, ".bls_pub_key"),
+                signature: stdJson.readBytes(registrationJson, ".signature"),
+                depositDataRoot: stdJson.readBytes32(registrationJson, ".deposit_data_root"),
+                blsEncryptedPrivKeyShares: blsEncryptedPrivKeyShares,
+                blsPubKeySet: stdJson.readBytes(registrationJson, ".bls_pub_key_set"),
+                raveEvidence: ""
+            });
+
+            Permit memory pufETHPermit;
+            pufETHPermit.amount = 2 ether;
+
+            Permit memory vtPermit;
+            vtPermit.amount = vtAmount * 1 ether;
+
+            bytes memory registerValidatorKeyCalldata =
+                abi.encodeCall(PufferProtocol.registerValidatorKey, (validatorData, moduleName, pufETHPermit, vtPermit));
+
+            transactions[i + 2] = Tx({ to: protocolAddress, data: registerValidatorKeyCalldata });
+
+            registeredPubKeys.push(validatorData.blsPubKey);
+        }
+
+        // Create Safe TX JSON
+        _createSafeJson(safe, transactions);
+
+        console.log("Validator PubKeys:");
+        for (uint256 i = 0; i < registeredPubKeys.length; ++i) {
+            console.logBytes(registeredPubKeys[i]);
+        }
+    }
+
+    function _createSafeJson(address safe, Tx[] memory transactions) internal {
+        // First we need to craft the JSON file for the transactions batch
+        string memory root = "root";
+
+        vm.serializeString(root, "version", "\"1.0\"");
+        vm.serializeUint(root, "createdAt", block.timestamp * 1000);
+        vm.serializeString(root, "chainId", "\"1\"");
```

Comment:
This needs to be generalized to work with Holesky (tested that changing to 17000 works perfectly)